### PR TITLE
chore(oas): replace response with $ref class JSDoc (Admin SHI-V)

### PR DESCRIPTION
**Pull request by patrick-medusajs**
### Scope

Admin routes directories SHI to V.

### What

Move inline OAS response schema declaration under their respective class declarations in order to expose them through  `#/components/schemas`. Replace inline OAS response schema with a `$ref` reference pointing to the newly declared schema.

### Why

Having response declared as its own "named" schema will allow OAS code generators to output typed entities/DTO that can be consumed without having to reference the route/operation.

### How

Declare a new @schema JSDoc for each "Res" class used to parse and validate request body. Move the current inline requestBody to the new @schema.

### Test

- Ran OAS validator.
- Ran docs build script.

Expect no visible changes to the documentation.

### CODE REVIEW

File: packages/medusa/src/api/routes/admin/shipping-profiles/index.ts [hunk 1]
```diff
@@ -41,15 +41,47 @@ export const defaultAdminShippingProfilesFields: (keyof ShippingProfile)[] = [
...
+ *   object:
+ *     type: string
+ *     description: The type of the object that was deleted.
+ *     default: shipping_profile
+ *   deleted:
+ *     type: boolean
+ *     description: Whether or not the items were deleted.
+ *     default: true
+ */
+export type AdminDeleteShippingProfileRes = DeleteResponse
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/shipping-profiles/index.ts:63]: simply moved below all relation declarations

File: packages/medusa/src/api/routes/admin/stock-locations/index.ts [hunk 2]
```diff
@@ -85,12 +85,35 @@ export const defaultAdminStockLocationFields: (keyof StockLocationDTO)[] = [
...
+ * type: object
+ * properties:
+ *   stock_location:
+ *     $ref: "#/components/schemas/StockLocationDTO"
+ */
 export type AdminStockLocationsRes = {
   stock_location: StockLocationDTO
 }
 
-export type AdminStockLocationsDeleteRes = DeleteResponse
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/stock-locations/index.ts:92]: No usage found across the codebase. Not implemented.

**Review by patrick-medusajs — COMMENTED** [review]
self-review: highlighting noteworthy code changes.

**Review by adrien2p — APPROVED** [review]

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Pull request merged**